webapp/src/pages/Users/Users.tsx

```diff
@@ -8,56 +8,43 @@ import {
   RowsListProps,
   SmallAvatar,
 } from '../../shared/components';
-import React, { useEffect, useState } from 'react';
 import {
   USER_URL,
   USERS_EP_URL,
   USERS_URL,
   WILDCARD_AVATAR_URL,
 } from '../../shared/urls';
 import { Link } from 'react-router-dom';
+import { useData } from '../../shared/hooks/UseData';
+import { instanceOfArrayTyped, instanceOfUser, User } from '../../shared/types';
 
-interface User {
-  readonly username: string;
-  readonly email: string;
-  readonly avatarId: string | null;
-  readonly id: string;
-  readonly createdAt: Date;
+function instanceOfUserArray(value: object): boolean {
+  return instanceOfArrayTyped(value, instanceOfUser);
 }
 
 export default function Users() {
-  const [usersList, setUsersList] = useState<RowsListProps>({ rows: [] });
-
-  const getRows = async (url: string): Promise<RowsListProps> => {
-    const users = (await fetch(url).then((response) => response.json())) ?? [];
+  const GetRows = (): RowsListProps => {
     return {
-      rows: users.map((user: User) => {
-        return {
-          iconVariant: IconVariant.ARROW_FORWARD,
-          avatarProps: {
-            url: user.avatarId
-              ? `${USERS_EP_URL}/${user.id}/avatar`
-              : WILDCARD_AVATAR_URL,
-            status: 'offline',
-          },
-          url: `${USERS_URL}/${user.username}`,
-          title: user.username,
-          subtitle: 'level x',
-          key: user.id,
-        };
-      }),
+      rows: useData<User[]>(USERS_EP_URL, [], instanceOfUserArray).map(
```

Comment:
I'm not sure it's a good idea to use this hook, as we may want to keep some state here for the search use feature. Also we'd have conflicts with search users form mr